editor/components/MapGeocoderControl.tsx

```diff
@@ -0,0 +1,124 @@
+import { useControl, ControlPosition } from "react-map-gl";
+import MapboxGeocoder, { GeocoderOptions } from "@mapbox/mapbox-gl-geocoder";
+import "@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css";
+import { LatLon } from "@/components/CrashMap";
+
+const MAPBOX_TOKEN = process.env.NEXT_PUBLIC_MAPBOX_TOKEN;
+
+type GeocoderControlProps = Omit<
+  GeocoderOptions,
+  "accessToken" | "mapboxgl" | "marker"
+> & {
+  position: ControlPosition;
+  onResult?: (latLon: LatLon) => void;
+};
+
+const ATX_BOUNDING_BOX: [number, number, number, number] = [
+  -98.182, 29.987, -97.304, 30.663,
+];
+
+/**
+ * Geocode control
+ * See: https://github.com/visgl/react-map-gl/tree/7.0-release/examples/geocoder
+ */
+export default function MapGeocoderControl(props: GeocoderControlProps) {
+  const geocoder = useControl<any>( // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
The typing on this needed some work after I copypasta'd the example. I couldn't get this bit to work and gave up. I think there is really low risk here given that this control clearly works...?

Reply:
ive been getting warnings about using the `any` type in a few places in our code. i think using it here is maybe an okay exception since it can be so hard to make types work w external libraries :shrug: 

curious though why doesnt `const geocoder = useControl<MapboxGeocoder>` work like in the copypasta'd example?